code/modules/mob/living/silicon/say.dm

```diff
@@ -10,24 +10,45 @@
 	if(issilicon(src))
 		var/mob/living/silicon/S = src
 		desig = trim_left(S.designation + " " + S.job)
-	var/message_a = say_quote(message)
-	var/rendered = "Robotic Talk, <span class='name'>[name]</span> <span class='message'>[message_a]</span>"
+	var/large_message_a = say_quote(message, list("robot big"))
+	var/message_a = say_quote(message, list("robot"))
+	var/mob/living/silicon/ai/true_ai_core
+	if(iscyborg(src))  // this detects if a borg is AI shell, so that they can be loud always
+		var/mob/living/silicon/robot/ai_shell = src
+		true_ai_core = ai_shell.mainframe
 	for(var/mob/M in GLOB.player_list)
 		if(M.binarycheck())
 			if(isAI(M))
-				var/renderedAI = "<span class='srt_radio binarysay'>Robotic Talk, <a href='?src=[REF(M)];track=[html_encode(name)]'><span class='name'>[name] ([desig])</span></a> <span class='message'>[message_a]</span></span>"
-				to_chat(M, renderedAI)
+				var/loud = FALSE
+				if(M == src)
+					loud = TRUE
+				var/rendered = "<span class='srt_radio binarysay'>Robotic Talk, <a href='?src=[REF(M)];track=[html_encode(name)]'><span class='name'>[name] ([isAI(M) ? "[desig]" : ""])</span></a> [loud ? "[large_message_a]" : "[message_a]"]</span>"
+				to_chat(M, rendered) //AI hears only itself on loud mode.
+			else if(iscyborg(M))
+				var/mob/living/silicon/robot/borg = M
+				var/loud = FALSE
+				if(true_ai_core == borg.connected_ai)
+					loud = TRUE
+				var/rendered = "<span class='srt_radio binarysay'>Robotic Talk, <span class='name'>[name]</span> [loud ? "[large_message_a]" : "[message_a]"]</span>"
+				to_chat(M, rendered) //Cyborg only hears master AI on loud mode.
 			else
-				to_chat(M, "<span class='srt_radio binarysay'>[rendered]</span>")
+				var/loud = FALSE
+				if(isAI(src))
+					loud = TRUE
+				var/rendered = "<span class='srt_radio binarysay'>Robotic Talk, <span class='name'>[name]</span> [loud ? "[large_message_a]" : "[message_a]"]</span>"
+				to_chat(M, rendered) //Human with binary channel hears all AIs on loud mode.
 		if(isobserver(M))
 			var/following = src
+			var/loud = FALSE
 			// If the AI talks on binary chat, we still want to follow
 			// it's camera eye, like if it talked on the radio
-			if(isAI(src))
+			if(isAI(src) || true_ai_core)
```

Comment:
```suggestion
			if(isAI(src))
```
umh, actually, `following` shouldn't be chaged when `true_ai_core` is true. my mistake.